torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -3653,14 +3653,21 @@ at::Tensor XLANativeFunctions::pixel_unshuffle(const at::Tensor& self,
       pixel_unshuffle)>::call(self, downscale_factor);
 }
 
-at::Tensor XLANativeFunctions::reshape_symint(const at::Tensor& self,
-                                              c10::SymIntArrayRef shape) {
+at::Tensor XLANativeFunctions::reshape(const at::Tensor& self,
+                                       at::IntArrayRef shape) {
   // See Note: [Disabling functionalization]
   if (runtime::sys_util::GetEnvBool("XLA_DISABLE_FUNCTIONALIZATION", false)) {
-    return at::native::reshape_symint(self, shape);
+    return at::native::reshape_symint(
+        self,
+        c10::SymIntArrayRef(reinterpret_cast<const c10::SymInt*>(shape.data()),
+                            shape.size()));
   }
+  auto inner_tensor = torch::lazy::maybe_unwrap_functional(self);
   return at::functionalization::functionalize_aten_op_symint<ATEN_OP(
```

Comment:
Hmm. I think we have to call `MaybeWrapTensorToFunctional` on the result of `functionalize_aten_op_symint` (a). The thing is that (a) wraps the tensor arguments with `FunctionalTensorWrapper` at first and, before returning, unwraps it. However, we do need to return a `FunctionalTensorWrapper` (see `AtenFromXlaTensor` function).

Reply:
Thanks for the pointer, I've updated the code to do that and seems like the warning is gone, but I'll wait for the CI to verify the unit tests.